identify/identify.py

```diff
@@ -159,6 +159,15 @@ def parse_shebang(bytesio):
     cmd = tuple(_shebang_split(first_line.strip()))
     if cmd and cmd[0] == '/usr/bin/env':
         cmd = cmd[1:]
+    if cmd and cmd[0].endswith('nix-shell'):
+        if bytesio.read(2) != b'#!':
+            return cmd
+        second_line = bytesio.readline().decode('UTF-8')
+        cmd = tuple(_shebang_split(second_line.strip()))
+        for i, token in enumerate(cmd):
+            if not token == '-i':
+                continue
+            cmd = (cmd[i + 1],)  # the argument to -i flag
```

Comment:
Indeed it will take the last one.

Reply:
What do you mean? Sorry, I couldn't make sense of this comment :confused: .

Reply:
The last occurrence of `-i`, so that's good.